compiler/rustc_infer/src/traits/engine.rs

```diff
@@ -52,8 +52,10 @@ pub trait TraitEngine<'tcx>: 'tcx {
 
     fn pending_obligations(&self) -> Vec<PredicateObligation<'tcx>>;
 
-    fn drain_obligations_stalled_on(&mut self, ty_var: ty::TyVid)
-    -> Vec<PredicateObligation<'tcx>>;
+    fn drain_unstalled_obligations(
+        &mut self,
+        infcx: &InferCtxt<'_, 'tcx>,
+    ) -> Vec<PredicateObligation<'tcx>>;
```

Comment:
This function deserves some documentation. I needed to look at both its body and its caller to understand what it does. Still not sure I do... leaving comments on the use site